DigitalLearningSolutions.Data/Services/CustomPromptsService.cs

```diff
@@ -72,7 +96,36 @@ public List<(int id, string value)> GetCustomPromptsAlphabeticalList()
             return customPromptsDataService.GetCustomPromptsAlphabetical().ToList();
         }
 
-        private static List<CustomPrompt> PopulateCustomPromptListFromCentreCustomPromptsResult(CentreCustomPromptsResult? result)
+        public bool AddCustomPromptToCentre(int centreId, int promptId, bool mandatory, string? options)
+        {
+            var centreCustomPrompts = GetCustomPromptsForCentreByCentreId(centreId);
+            var existingPromptNumbers = centreCustomPrompts.CustomPrompts
+                .Select(c => c.CustomPromptNumber)
+                .OrderBy(n => n);
+
+            var promptNumbers = new List<int> { 1, 2, 3, 4, 5, 6 };
+            promptNumbers.RemoveAll(x => existingPromptNumbers.Contains(x));
+            var promptNumber = promptNumbers.Any() ? promptNumbers.Min() : (int?)null;

```

Comment:
I'm wondering if we could use "Except" syntax here: https://docs.microsoft.com/en-us/dotnet/api/system.linq.enumerable.except?view=netcore-3.1

Also, do we need to order the "existingPromptNumbers if we do .Min() later to get the promptNumber?